Update Spring Data JPA With Hibernate User's Guide for Altibase.md 영문 (#505)
ALTIBASE/Documents: 3rd Party Guide for Altibase/eng/Spring Data JPA With Hibernate 6.4 User's Guide for Altibase.md

- Spring Boot 3.2.2
- Spring Data JPA 3.2.2
- Java 17
- Altibase server 7.1.0.9.3 or above
- Altibase JDBC driver 7.1.0.9.0 or above

<br/>

# Integrate Altibase With Spring Boot

## 1. Generate Spring Boot Project With Spring initializr

Visit [https://start.spring.io/](#https://start.spring.io/), then enter the following information. After that, click on "Generate" to create the project. Click on "ADD DEPENDENCIES..." under Dependencies, and add "Spring Web" and "Spring Data JPA". For the convenience of writing example code, also add "Lombok".

<img src="Images/JPA/spring_initializr.png"/>

## 2. Add Dependencies to pom.xml

### Add AltibaseDialect Dependency

Starting from Hibernate 6.4, the AltibaseDialect has been added to hibernate-community-dialect. Therefore, add a dependency on "hibernate-community-dialect" with a version of 6.4 or above. (If users are using Hibernate version under 6.4, users need to manually compile and add the AltibaseDialect.)

```xml
<dependency>
    <groupId>org.hibernate.orm</groupId>
    <artifactId>hibernate-community-dialects</artifactId>
    <version>6.4.1.Final</version>
</dependency>
```

### Add Altibase JDBC Driver Dependency

> From patch version Altibase 7.1.0.9.0 for Altibase 7.1 and Altibase 7.3.0.0.2 for Altibase 7.3, the Altibase JDBC driver can be downloaded from the [Maven Central Repository](https://mvnrepository.com/artifact/com.altibase/altibase-jdbc). Add to Altibase JDBC driver dependency as follows: 

Reviewer: ```suggestion
From patch version Altibase 7.1.0.9.0 for Altibase 7.1 and Altibase 7.3.0.0.2 for Altibase 7.3, the Altibase JDBC driver can be downloaded from the [Maven Central Repository](https://mvnrepository.com/artifact/com.altibase/altibase-jdbc). Add the following to the Altibase JDBC driver dependency: 
```

* Add Altibase JDBC driver dependency for Altibase 7.3

  ```xml
  <dependency>
      <groupId>com.altibase</groupId>
      <artifactId>altibase-jdbc</artifactId>
      <version>7.3.0.0.2</version>
  </dependency>
  ```

* Add Altibase JDBC driver dependency for Altibase 7.1

  ```xml
  <dependency>
      <groupId>com.altibase</groupId>
      <artifactId>altibase-jdbc</artifactId>
      <version>7.1.0.9.2</version>
  </dependency>
  ```

## 3. Configure the application.properties

### Altibase JDBC Driver Configuration

```java
# Database
spring.datasource.driver-class-name=Altibase.jdbc.driver.AltibaseDriver
spring.datasource.url=jdbc:Altibase://localhost:20300/mydb
spring.datasource.username=sys
spring.datasource.password=manager

# JPA
spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation=true
```

> [!NOTE]
>
> **spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation=true** 
>

> > This configuration is to suppress the java.sql.SQLFeatureNotSupportedException error that may occur when Hibernate calls Connection.createNClob() to check if the NClob type is supported or not. Altibase does not support the NClob type, which leads to the occurrence of this error. However, it can be safely ignored.

Reviewer: ```suggestion
> This configuration is to suppress the java.sql.SQLFeatureNotSupportedException error that may occur when Hibernate calls Connection.createNClob() to check whether or not the NClob type is supported. Altibase does not support the NClob type, which causes this error to occur. However, it can be safely ignored.
```

> #### Altibase JDBC Driver Connection Configuration

Reviewer 2: connection attributes 이 나을것 같네요.  연결 속성

Author: 반영하였습니다~

> If users use the Altibase 7.1 JDBC driver, users need to add the connection attribute configuration as follows. The Altibase 7.3 JDBC driver users can skip this step.

Reviewer: ```suggestion
If users are using the Altibase 7.1 JDBC driver, they need to add the connection attribute configuration as follows. The Altibase 7.3 JDBC driver users can skip this step.
```

```java
... 
2024-02-23T16:33:03.171+09:00  INFO 48644 --- [           main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: default]
2024-02-23T16:33:03.252+09:00  INFO 48644 --- [           main] org.hibernate.Version                    : HHH000412: Hibernate ORM core version 6.4.1.Final
2024-02-23T16:33:03.294+09:00  INFO 48644 --- [           main] o.h.c.internal.RegionFactoryInitiator    : HHH000026: Second-level cache disabled
2024-02-23T16:33:03.582+09:00  INFO 48644 --- [           main] o.s.o.j.p.SpringPersistenceUnitInfo      : No LoadTimeWeaver setup: ignoring JPA class transformer
2024-02-23T16:33:03.614+09:00  INFO 48644 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2024-02-23T16:33:03.706+09:00  INFO 48644 --- [           main] com.zaxxer.hikari.pool.HikariPool        : HikariPool-1 - Added connection Altibase.jdbc.driver.AltibaseConnection@2a99ca99
2024-02-23T16:33:03.708+09:00  INFO 48644 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
...
```

# Example

> This section provides a simple example that includes creating the entity class named Book.java under the com.example.AltitestJPA package, running the application, and checking whether the table is automatically created.

Reviewer: ```suggestion
This section provides a simple example that involves creating an entity class called Book.java in the com.example.AltitestJPA package, running the application, and checking that the table is created automatically.
```
